plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -34,15 +36,23 @@ class Init {
 	 */
 	private $redirection_controller;
 
+	/**
+	 * Simple product block template.
+	 *
+	 * @var AbstractBlockTemplate
+	 */
+	public $simple_product_template;
+
 	/**
 	 * Constructor
 	 */
 	public function __construct() {
 		if ( Features::is_enabled( 'product-variation-management' ) ) {
-			array_push($this->supported_post_types, 'variable');
+			array_push( $this->supported_post_types, 'variable' );
 		}
 
-		$this->redirection_controller = new RedirectionController( $this->supported_post_types );
+		$this->simple_product_template = new SimpleProductTemplate();
```

Comment:
This was done so that Plugins can still update the `SimpleProductTemplate` before the `woocommerce_register_post_type_product` hook is called, if I instantiate it right in `add_product_template` there is no way for a plugin to modify it.
Of-course this logic will be short lived as it will become part of the registry.

Reply:
Since this will change when we use the registry, I'm okay with this.